scipy/stats/_multivariate.py

```diff
@@ -3521,3 +3522,353 @@ def rvs(self, eigs, random_state=None, tol=1e-13, diag_tol=1e-7):
+shape : positive scalar, optional
+    Shape of the distribution (default one)
+scale : positive scalar, optional
+    Scale on inverse gamma distribution prior (default one)
+"""
+
+_nig_doc_callparams_note = \
+    """The parameters `loc` and `variance_scale` serves as the
+    hyperparameters for normal distribution which models the
+    unknown mean. The parameters `shape` and `scale` serves as
+    the hyperparameters for inverse gamma distribution which
+    models the unknown variance.
+    """
+
+_nig_doc_frozen_callparams = ""
+
+_nig_doc_frozen_callparams_note = \
+    """See class definition for a detailed description of parameters."""
+
+nig_docdict_params = {
+    '_nig_doc_default_callparams': _nig_doc_default_callparams,
+    '_nig_doc_callparams_note': _nig_doc_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+nig_docdict_noparams = {
+    '_nig_doc_default_callparams': _nig_doc_frozen_callparams,
+    '_nig_doc_callparams_note': _nig_doc_frozen_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+
+class normal_invgamma_gen(multi_rv_generic):
+    r"""
+    A normal inverse gamma random variable.
+
+    Methods
+    -------
+    ``pdf(x, sig2, loc=0, variance_scale=1, shape=1, scale=1)``
+        Probability density function.
+    ``logpdf(x, sig2, loc=0, variance_scale=1, shape=1, scale=1)``
+        Log of the probability density function.
+    ``rvs(loc=0, variance_scale=1, shape=1, scale=1, size=1, random_state=None)``
+        Draw random samples from normal and inverse gamma distribution.
+    ``mean(loc=0, variance_scale=1, shape=1, scale=1)``
+        Mean of the random variates.
+    ``mode(loc=0, variance_scale=1, shape=1, scale=1)``
+        Mode of the random variates.
+
+    Parameters
+    ----------
+    x : array
+            One-dimensional array.
+    sig2 : array
+            One-dimensional array.
+    %(_nig_doc_default_callparams)s
+    %(_doc_random_state)s
+
+    Alternatively, the object may be called (as a function) to fix the
+    loc, variance_scale, shape, scale parameters, returning a "frozen"
+    normal inverse gamma random variable:
+
+    rv = normal_invgamma(loc=0, variance_scale=1, shape=1, scale=1)
+        - Frozen object with the same methods but holding the given
+          loc, variance_scale, shape, scale.
+
+    Notes
+    -----
+    %(_nig_doc_callparams_note)s
+
+    The probability density function for `normal_invgamma` is
+
+    .. math::
+
+        f(x,\sigma^2) = \frac {1} {\sigma\sqrt{2\pi\nu} } \, \frac{\beta^\alpha}
+                        {\Gamma(\alpha)} \, \left( \frac{1}{\sigma^2} \right)^
+                        {\alpha + 1}\exp \left( \frac { -\beta}{\sigma^2} -
+                        \frac{(x - \mu)^2} {2\sigma^2\nu}  \right),
+
+    where :math:`\mu` is the loc, :math:`\nu` the variance scale,
+    :math:`\alpha` the shape, :math:`\beta` the scale and :math:`x` and
+    `\sigma^2` takes values.
```

Comment:
We typically encourage meaningful names. However here, I'm not sure "shape", "loc" and "variance_scale"  are worth it (besides, "shape" is used for general shape parameters for univariate distributions). Maybe rename them back to `alpha`, `beta` and `nu` in function signatures and documentation.